src/core/postAttachments.ts

```diff
@@ -39,14 +39,43 @@ export class Attachments {
                     ],
                 );
 
-                attachmentsProp.push({contentId: contentId, id: resource.id});
+                attachmentsProp.push({contentId: contentId, id: resource.id, fileName: filename, mimeType: mimeType});
             }
 
             return attachmentsProp;
         } catch (err) {
             throw err;
         }
     }
+
+    async postInlineAttachment(attachment: Attachment): Promise<AttachmentProperties> {
+        const tempFolder = this.tempFolderPath;
+
+        try {
+            // for each attachment will create an actual file of the attachment and posting to Joplin.
+            const {contentId, filename, mimeType, content} = attachment;
+            const path = `${tempFolder}${filename}`;
+
+            // to create a file
+            fs.writeFileSync(path, Buffer.from(content));
+
+            // To post a file to Joplin
+            const resource = await joplin.data.post(
+                ['resources'],
+                null,
+                {title: filename}, // Resource metadata
+                [
+                    {
+                        path: path, // Actual file
+                    },
+                ],
+            );
+
+            return {contentId: contentId, id: resource.id, fileName: filename, mimeType: mimeType};
+        } catch (err) {
```

Comment:
no point in catching if you re-throw the error immediately again

Reply:
This is true. I removed it now.